BusinessAdapter.FP.UnitTests/Receiving/FpMessageReceiverTests.cs

```diff
@@ -0,0 +1,164 @@
+			{
+				Directory = string.Empty,
+				MessageLimitCount = 10,
+				Retry = new Schleupen.AS4.BusinessAdapter.Configuration.RetryOption { Count = 3 }
+			});
+			// Act & Assert
+			Assert.That(async () => await fpMessageReceiver.ReceiveAvailableMessagesAsync(CancellationToken.None),
+				Throws.TypeOf<CatastrophicException>()
+					.With.Message.EqualTo("The receive directory is not configured."));
+		}
+
+		[Test]
+		public void ReceiveAvailableMessagesAsync_ThrowsCatastrophicException_WhenNoMarketPartnersAreConfigured()
+		{
+			adapterOptionsMock.Value.Marketpartners = [];
+			receiveOptionsMock = Options.Create<ReceiveOptions>(new ReceiveOptions
+			{
+				Directory = "C:\\Adapter\\Receive",
+				MessageLimitCount = 10,
+				Retry = new Schleupen.AS4.BusinessAdapter.Configuration.RetryOption { Count = 3 }
+			});
+			fpMessageReceiver = new FpMessageReceiver(
+				loggerMock.Object,
+				receiveOptionsMock,
+				adapterOptionsMock,
+				businessApiGatewayFactoryMock.Object,
+				fpFileRepositoryMock.Object,
+				eicMappingMock);
+
+			// Act & Assert
+			Assert.That(async () => await fpMessageReceiver.ReceiveAvailableMessagesAsync(CancellationToken.None),
+				Throws.TypeOf<CatastrophicException>()
+					.With.Message.EqualTo("No valid own market partners were found."));
+		}
+
+		[Test]
+		public void ReceiveAvailableMessagesAsync_ShouldTrackMarketPartnersWithoutCertificate()
+		{
+			// Arrange
+			receiveOptionsMock = Options.Create<ReceiveOptions>(new ReceiveOptions
+			{
+				Directory = "C:\\Adapter\\Receive",
+				MessageLimitCount = 10,
+				Retry = new Schleupen.AS4.BusinessAdapter.Configuration.RetryOption { Count = 3 }
+			});
+			fpMessageReceiver = new FpMessageReceiver(
+				loggerMock.Object,
+				receiveOptionsMock,
+				adapterOptionsMock,
+				businessApiGatewayFactoryMock.Object,
+				fpFileRepositoryMock.Object,
+				eicMappingMock);
+			var gatewayMock = new Mock<IBusinessApiGateway>();
+			var messages = new[] { CreateFpInboxMessage() };
+			var receiveInfo = new MessageReceiveInfo(messages);
+
+			businessApiGatewayFactoryMock
+				.Setup(factory => factory.CreateGateway(It.IsAny<FpParty>()))
+				.Returns(gatewayMock.Object);
+
+			gatewayMock
+				.Setup(g => g.QueryAvailableMessagesAsync(It.IsAny<int>()))
+				.ThrowsAsync(new Schleupen.AS4.BusinessAdapter.Certificates.MissingCertificateException("TestMarketPartner"));
+
+			// Act
+			var ex = Assert.ThrowsAsync<AggregateException>(async () =>
+				await fpMessageReceiver.ReceiveAvailableMessagesAsync(CancellationToken.None));
+
+			// Assert
+			var innerException = ex!.InnerExceptions[0];
+			Assert.That(innerException, Is.InstanceOf<Schleupen.AS4.BusinessAdapter.Certificates.MissingCertificateException>());
+			Assert.That(((Schleupen.AS4.BusinessAdapter.Certificates.MissingCertificateException)innerException).MarketpartnerIdentificationNumber,
+				Is.EqualTo("TestMarketPartner"));
+		}
+

```

Comment:
we only have test cases for error situations. Missing testcase ...
- check if ReceiveStatus is correclty if all messages are successful received
- check if ReceiveStatus is correclty if some messages received correctly and some are received with failures -> that should not block new message stream
- ...